### PR TITLE
Add onetime charge as usable service to HA

### PR DESCRIPTION
**Pull request by woehrl01**
This PR adds a onetime charge service to HA, to trigger a onetime water heating cycle.

### CODE REVIEW

File: vicare/water_heater.py [hunk 1]
```diff
@@ -43,13 +52,13 @@
 }
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
```
Comment (oischinger) [inline vicare/water_heater.py:55]: I belive we were not able to use async because we do I/O when restoring the token.
Did that behavior change in any way?

Comment (woehrl01) [reply vicare/water_heater.py:55]: Good point, I haven't thought of that.

Comment (oischinger) [reply vicare/water_heater.py:55]: works nicely unless the token expires :D

Comment (woehrl01) [reply vicare/water_heater.py:55]: You're right 👍🏽

Comment (woehrl01) [reply vicare/water_heater.py:55]: Just had a more in-depth look, this is only setting up the platform, as there is no API called, this is safe to call async. Actually, `climate.py` is already async.

**Comment by blehers** [comment]
This function would be great! 

Have you noticed that at least for Vitocal 300G the temperature 2 is used. A test response of this type is available at [PyViCare](https://github.com/somm15/PyViCare/blob/master/tests/test_Vitocal300G.py).

[Viessmann manual (page 29)](https://www.manualslib.com/manual/533263/Viessmann-Vitocal-300-G.html?page=29#manual):
> You can specify a second temperature
> (Set temperature 2) for DHW heating.
> ■ By selecting the operating status
> "Temp. 2" in the DHW time program,
> you can heat the DHW to that set value
> (see page 31).
> ■ The "Set temperature 2" is the set
> value for heating the DHW once (see
> page 33) and for the manual opera-
> tion (see page 45).

**Comment by woehrl01** [comment]
Hi @oischinger I just rebased the changes on the latest integration.

**Comment by oischinger** [comment]
Thanks

**Pull request merged**